consensus/state.go

```diff
@@ -630,6 +630,13 @@ func (ms *MidState) fileContractElement(ts V1TransactionSupplement, id types.Fil
 	return ts.revision(id)
 }
 
+func (ms *MidState) storageProof(ts V1TransactionSupplement, id types.FileContractID) (V1StorageProofSupplement, bool) {
+	if i, ok := ms.created[id]; ok {
+		return V1StorageProofSupplement{FileContract: ms.fces[i], WindowID: ms.base.Index.ID}, true
```

Comment:
Always returning true doesn't seem right either. If there was ever a bug that managed to call Apply without Validate, it could silently finalize contracts instead of panicking. I agree that the error is unhelpful though. Maybe storageProof could return an error instead?

Reply:
tbf there are a *lot* of things that can go wrong, loudly or silently, if you call Apply without Validate.

In this case I think the problem comes from having a single `storageProof` method instead of splitting it into `fileContract` and `windowID`. I can fix that.